### PR TITLE
Add missing properties in FastifyRequest and FastifyReply

### PR DESCRIPTION
**Pull request by cpylua**
- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Ethan-Arrowood** [comment]
Are these properties not apart of the IncommingMessage interface from Node.js?

**Comment by cpylua** [comment]
> Are these properties not apart of the IncommingMessage interface from Node.js?

Fastify's request object doesn't inherit Node.js raw request according to the code.
https://github.com/fastify/fastify/blob/master/lib/request.js

I think `FastifyRequest` should not extend `RawRequest` here.

```typescript
export type FastifyRequest<
  RawServer extends RawServerBase = RawServerDefault,
  RawRequest extends RawRequestDefaultExpression<RawServer> = RawRequestDefaultExpression<RawServer>,
  RequestGeneric extends RequestGenericInterface = RequestGenericInterface
> = RawRequest & FastifyRequestInterface<RawServer, RawRequest, RequestGeneric>
```

**Comment by zmeyc** [comment]
@cpylua +1, seems it shouldn't extend RawRequest. `connection` field is also missing. I propose sorting TS fields to match order of field declarations in request.js to spot omissions easier.

**Comment by cpylua** [comment]
@Ethan-Arrowood please have another look.

**Comment by mcollina** [comment]
You always got it wrong. Fastify request is our object. request.raw is IncomingMessage.

**Comment by cpylua** [comment]
Made some small changes to the doc regarding `FastifyRequest`.

I just realized `FastifyReply` definition has the same bug with `FastifyRequest`. I will have a try next week.

File: types/reply.d.ts [hunk 1]
```diff
@@ -1,42 +1,52 @@
...
 }
 
 export type FastifyReply<
   RawServer extends RawServerBase = RawServerDefault,
+  RawRequest extends RawRequestDefaultExpression<RawServer> = RawRequestDefaultExpression<RawServer>,
   RawReply extends RawReplyDefaultExpression<RawServer> = RawReplyDefaultExpression<RawServer>,
+  RequestGeneric extends RequestGenericInterface = RequestGenericInterface,
   ContextConfig = ContextConfigDefault
-> = RawReply & FastifyReplyInterface<RawServer, RawReply, ContextConfig>
+> = FastifyReplyInterface<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>
```
Comment (Ethan-Arrowood) [inline types/reply.d.ts:52]: Looking at this again, I believe it is unnecessary to have this type any longer. The point of breaking them apart was in order to support both declaration merging and generic inheritance. Now that the FastifyRequest and FastifyReply  types are no longer extending from Node.js, but only using it in the `raw` field, I say we rename some things.

I think the best path is to remove this type (and FastifyRequest as well), and then rename the respective interfaces to `FastifyReply` and `FastifyRequest`.

Comment (jcao219) [reply types/reply.d.ts:52]: +1 on just using `FastifyReply` and `FastifyRequest` idea

File: types/reply.d.ts [hunk 2]
```diff
@@ -1,42 +1,52 @@
...
+  callNotFound(): void;
   getResponseTime(): number;
-  type(contentType: string): FastifyReply<RawServer, RawReply>;
-  context: FastifyContext<ContextConfig>;
-  raw: RawReply;
-  sent: boolean;
+  type(contentType: string): FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
+  serializer(fn: (payload: any) => string): FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
+  serialize(payload: any): string;
+  then(fullfilled: () => void, rejected: (err: Error) => void): void;
```
Comment (jcao219) [inline types/reply.d.ts:43]: shouldn't this return another `thenable`?

Comment (cpylua) [reply types/reply.d.ts:43]: You're right about the spec, but I think it is what the code does right now.
https://github.com/fastify/fastify/blob/master/lib/reply.js#L299

File: types/reply.d.ts (hunk 1 above)
Comment (cpylua) [reply types/reply.d.ts]: Done

Comment (Ethan-Arrowood) [reply types/reply.d.ts]: great! I'll check it tomorrow

File: test/types/serverFactory.test-d.ts [hunk 3]
```diff
@@ -28,11 +28,11 @@ const serverFactory: FastifyServerFactory<http.Server> = (handler, opts) => {
 const customServer = fastify<http.Server, CustomIncomingMessage, CustomServerResponse>({ serverFactory })
 
 customServer.get('/', function (request, reply) {
-  if (request.fakeMethod) {
-    expectType<CustomType>(request.fakeMethod())
+  if (request.raw.fakeMethod) {
```
Comment (mcollina) [inline test/types/serverFactory.test-d.ts:31]: why was this moved to `request.raw`?

File: types/hooks.d.ts [hunk 4]
```diff
@@ -30,7 +30,7 @@ export interface onRequestHookHandler<
 > {
   (
     request: FastifyRequest<RawServer, RawRequest, RequestGeneric>,
-    reply: FastifyReply<RawServer, RawReply, ContextConfig>,
+    reply: FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>,
```
Comment (mcollina) [inline types/hooks.d.ts:33]: Why do we need both `RawRequest` and `RequestGeneric`? Aren't those two the same?

File: test/types/serverFactory.test-d.ts (hunk 3 above)
Comment (cpylua) [reply test/types/serverFactory.test-d.ts:31]: because `FastifyRequest` doesn't inherit nodejs request any more.

File: types/hooks.d.ts (hunk 4 above)
Comment (cpylua) [reply types/hooks.d.ts:33]: No.`RawRequest` is nodejs request and `RequestGeneric` is user defined types for request body, querystring, params and headers.

File: types/reply.d.ts (hunk 2 above)
Comment (Ethan-Arrowood) [reply types/reply.d.ts:43]: should open a new issue about this if this is a concern
